src/plugins/explore/public/components/visualizations/table/data_link_options.tsx

```diff
@@ -0,0 +1,333 @@
+              <EuiPopover
+                button={
+                  <EuiButtonIcon
+                    iconType="plusInCircle"
+                    onClick={() => setIsPopoverOpen(true)}
+                    aria-label="Add Field"
+                    data-test-subj="dataLinkAddFieldButton"
+                  />
+                }
+                isOpen={isPopoverOpen}
+                closePopover={() => setIsPopoverOpen(false)}
+                panelPaddingSize="s"
+                anchorPosition="downLeft"
+              >
+                <EuiContextMenu
+                  size="s"
+                  initialPanelId={0}
+                  panels={fieldOptions}
+                  data-test-subj="dataLinkFieldContextMenu"
+                />
+              </EuiPopover>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiFormRow
+          label={i18n.translate('explore.stylePanel.table.dataLinks.urlLabel', {
+            defaultMessage: 'URL',
+          })}
+          helpText={i18n.translate('explore.stylePanel.table.dataLinks.urlHelpText', {
+            defaultMessage: 'Use {cellValue} to include the cell value in the URL',
+            values: {
+              cellValue: '${__value.text}',
+            },
+          })}
+        >
+          <DebouncedFieldText
+            value={url}
+            placeholder="Enter data link url"
+            onChange={(value: string) => setUrl(value)}
+            data-test-subj="dataLinkUrlInput"
+          />
+        </EuiFormRow>
+        <EuiFormRow>
+          <EuiSwitch
+            compressed
+            label={i18n.translate('explore.stylePanel.table.dataLinks.openInNewTabLabel', {
+              defaultMessage: 'Open in new tab',
+            })}
+            checked={openInNewTab}
+            onChange={(e) => setOpenInNewTab(e.target.checked)}
+            data-test-subj="dataLinkNewTabSwitch"
+          />
+        </EuiFormRow>
+      </EuiModalBody>
+      <EuiModalFooter>
+        <EuiSmallButtonEmpty onClick={onCancel}>
+          {i18n.translate('explore.stylePanel.table.dataLinks.cancelButton', {
+            defaultMessage: 'Cancel',
+          })}
+        </EuiSmallButtonEmpty>
+        <EuiSmallButton
+          onClick={handleSave}
+          fill
+          disabled={isSaveDisabled}
+          data-test-subj="dataLinkSaveButton"
+        >
+          {i18n.translate('explore.stylePanel.table.dataLinks.saveButton', {
+            defaultMessage: 'Save',
+          })}
+        </EuiSmallButton>
+      </EuiModalFooter>
+    </EuiModal>
+  );
+};
+
+export const DataLinkOptions: React.FC<TableVisStyleControlsProps> = ({
+  styleOptions,
+  onStyleChange,
+  numericalColumns = [],
+  categoricalColumns = [],
+  dateColumns = [],
+}) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [editingLink, setEditingLink] = useState<DataLink | undefined>(undefined);
+
+  const availableFields = [
```

Comment:
Nit: seems this will be passed to `DataLinkModal` as a dependency of `useMemo`, can we wrap it with useMemo?